### PR TITLE
Tasks list

### PR DESCRIPTION
**Pull request by godziatkowski**

### CODE REVIEW

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package pl.pszczolkowski.mustdo.domain.task.bo;
+
+import pl.pszczolkowski.mustdo.domain.task.dto.TasksListSnapshot;
+
+public interface TasksListBO {
+
+   TasksListSnapshot add(String name, Long boardId);
+
+   TasksListSnapshot rename(Long id, String name);
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java:9]: Due to CQS rule, rename method should not return object that is modified

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+   
+   void delete(Long id);
+
+   public int removeTasksListsWithBoardId(Long boardId);
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java]: Be consistent - you don't use public modifier in other methods, so don't use it here

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/entity/TasksList.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+   @Id
+   @GeneratedValue(strategy = GenerationType.AUTO)
+   private Long id;
+
+   @NotNull
+   private Long boardId;
+
+   @NotNull
+   @Size(min = 2,
+      max = 100)
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/entity/TasksList.java]: Why you keep formatting @Size annotation like this? It's awful :C

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/entity/TasksList.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+   @Column(unique = true)
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/entity/TasksList.java]: It's invalid. TaskList name should be unique, but not among all TaskLists but among TasksLists in one Board

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/exception/TasksListAlreadyExistException.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package pl.pszczolkowski.mustdo.domain.task.exception;
+
+public class TasksListAlreadyExistException
+   extends RuntimeException {
+
+   private static final long serialVersionUID = -1794537307362685773L;
+
+   public TasksListAlreadyExistException() {
+      super("Task list with such name already exists");
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/exception/TasksListAlreadyExistException.java]: Maybe some additional information that list with that name exists but in the board would be better

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/finder/TasksListSnapshotFinder.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.List;
+
+import pl.pszczolkowski.mustdo.domain.task.dto.TasksListSnapshot;
+
+public interface TasksListSnapshotFinder {
+
+   TasksListSnapshot findOneById(Long id);
+
+   List<TasksListSnapshot> findAllWithBoardId(Long boardId);
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/finder/TasksListSnapshotFinder.java]: Uncle Bob says (and I agree with him) that method names in the same context (and preferably in whole project) should be consistent. So it should be "findBy..."

File: src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListApi.java [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+      @ApiResponse(code = 200,
+         message = "Found Tasks Lists for board with id")})
+   @RequestMapping(method = GET)
+   public HttpEntity<List<TasksList>> boardList(
+      @RequestParam(value = "boardId",
+         required = true) Long boardId) {
+
+      List<TasksList> tasksLists = tasksListSnapshotFInder
+         .findAllWithBoardId(boardId)
+         .stream().map(TasksList::new).collect(Collectors.toList());
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListApi.java]: Split it to separate lines

File: src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListNew.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+   public String getName() {
+      return name;
+   }
+
+   public void setName(String name) {
+      this.name = name;
+   }
+
+   @ApiModelProperty(
+      value = "Unique identifier of Board to whcich Tasks List belongs",
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListNew.java]: whcich

File: src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListNew.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+public class TasksListNew {
+
+   @NotNull
+   @Size(min = 3,
+      max = 100)
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListNew.java]: Formatting :(

File: src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListRename.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+public class TasksListRename {
+
+   @NotNull
+   private Long id;
+   @NotNull
+   @Size(min = 2,
+      max = 100)
```
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListRename.java]: Formatting :(

File: src/main/resources/db/migration/V0.0.2_2015.10.24_01_01__tasks_list_table_created.sql [hunk 11]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE tasks_list (
+	id BIGINT auto_increment,
+	NAME VARCHAR(100),
+	board_id BIGINT,
+	KEY (id)
```
Comment (pszczolkowski) [inline src/main/resources/db/migration/V0.0.2_2015.10.24_01_01__tasks_list_table_created.sql]: I might be wrong, but I believe that KEY is not the same as PRIMARY KEY in MySQL

File: src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+@SpringApplicationConfiguration(classes = Application.class)
+@WebAppConfiguration
+@Transactional
+@TransactionConfiguration(defaultRollback = true)
+public class TasksListBOTest extends ScenarioTest<GivenTasksListBO, WhenTasksListBO, ThenTasksListBO> {
+	private static final String CLAZZ = BoardBO.class.getSimpleName();
+
+	@Autowired
+	@ProvidedScenarioState
+	private BoardBO boardBO;
```
Comment (pszczolkowski) [inline src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java:36]: Do not mix domains! These are test cases for task domain, so you should not use any part of board domain here

File: src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+	
+	@After
+	public void tearDown(){
+		tasksListRepository.deleteAll();
+		boardRepository.deleteAll();
+	}
+	
+	@Test
+	public void should_add_new_tasksList(){
+		given().a_board()
```
Comment (pszczolkowski) [inline src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java:58]: This is test case for task domain, don't create entities from other domains. Change it in other tests also

File: src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+		then().tasksList_should_be_renamed();
+	}
+	
+	@Test
+	public void should_throw_tasksListAlreadyExistException_when_rename_invoked_and_tasksList_already_exist(){
+		given().a_board()
+			.and().a_tasksList_with_name(CLAZZ)
+			.and().other_tasksList_with_name(CLAZZ+"other");
+		when().rename_is_invoked_with_name(CLAZZ);
+		then().should_throw_tasksListAlreadyExistException();
```
Comment (pszczolkowski) [inline src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java:87]: Method name: there's missing subject. You should use passive instead, so it should be "tasksListAlreadyExistException_should_Be_thrown"

File: src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+			.and().a_tasksList_with_name(CLAZZ);
+		when().delete_is_invoked();
+		then().board_should_be_removed();
+	}
+   
+   @Test
+   public void should_delete_tasksListsWithGivenBoardId(){
+      given().a_board().and().few_tasks_list_with_boardId();
+      when().delete_by_boardId_invoked();
+      then().should_delete_tasksLists();
```
Comment (pszczolkowski) [inline src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java]: tasksLists should be deleted

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java (hunk 1 above)
Comment (godziatkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java:9]: Yeah and when I told you that in GrowUp project you said that entity isn't returning anything so it's ok....

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/entity/TasksList.java (hunk 3 above)
Comment (godziatkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/entity/TasksList.java]: Autoformating - I could change this but in my opinion it's not so bad

File: src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListNew.java (hunk 9 above)
Comment (godziatkowski) [inline src/main/java/pl/pszczolkowski/mustdo/web/restapi/tasksList/TasksListNew.java]: You are so annoing with this "Formatting" that i changed that.... Btw i was using the project style formatting....

File: src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java (hunk 12 above)
Comment (godziatkowski) [inline src/test/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBOTest.java:36]: THat's not me :)

File: src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java (hunk 1 above)
Comment (pszczolkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java:9]: When quoting my words, try to quote them more accurately. Anyway change this method so it won't return any snapshot

Comment (godziatkowski) [inline src/main/java/pl/pszczolkowski/mustdo/domain/task/bo/TasksListBO.java:9]: No.

**Pull request merged**